lib/logidze/versioned_association.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+module Logidze
+  module VersionedAssociation
+    def load_target
+      target = super
+
+      return target if inversed
+
+      time = owner.logidze_requested_ts
+
+      if target.is_a? Array
+        target.map! do |object|
+          object.at(time)
+        end.compact!
+      else
+        target.at!(time)
+      end
+
+      target
+    end
+
+    def stale_target?
+      logidze_stale? || super
+    end
+
+    def logidze_stale?
+      return false if !loaded? || inversed
+
+      unless target.is_a?(Array)
+        return owner.logidze_requested_ts != target.logidze_requested_ts
+      end
+
+      return false if target.empty?
+
+      owner.logidze_requested_ts != target.first.logidze_requested_ts
+    end
+
+    module CollectionAssociation
+      def ids_reader
+        reload
```

Comment:
Shouldn't we check whether association has been loaded? If it was, then we don't need to reload, do we? The same goes for `empty?`.